### ISSUE TITLE
MayBeConstant on @JvmField annotated property

### ISSUE THREAD

**Issue by realdadfish** [detekt/detekt#8640]
This was previously an issue on 1.23.x and still is on 2.0.0-alpha.0.

Consider the following code that mimicks an Android `BuildConfig` class that is generated compile time for testing purposes:

```kotlin
@Suppress("MayBeConstant")
object BuildConfigWithEndpoint {
    @JvmField
    val VERSION_NAME = "1.0"
    ...
}
```

Without the suppression, Detekt wants me to make `VERSION_NAME` constant, which is however prohibited when`@JvmField` is applied to that property:

```
JvmField cannot be applied to const property.
```

**Comment by BraisGabin** [comment]
This should be an easy one. It's ready for someone to grab it.

**Comment by atulgpt** [comment]
Hi @realdadfish can't we use the below code instead?
```kt
object BuildConfigWithEndpoint {
    const val VERSION_NAME = "1.0"
    ...
}
```

**Comment by marschwar** [comment]
Interestingly, the very first version of MayBeConst already included this test case:

```kotlin
		it("is a simple JvmField val") {
			val code = """
				@JvmField val x = 1
				"""
			subject.lint(code)
			assertThat(subject.findings).hasSize(1)
		}
```

I cannot think of any circumstance where this would be correct though.

**Comment by BraisGabin** [comment]
Why would you prefer `object A { @JvmStatic val a = 1 }` over `object A { const val a = 1 }`? It's true, the first generates the getter instead of making public the field. But that's exactly what this rule is about. It's a performance rule and it is doing its job.

I mean, if you don't agree with it you can or disable the rule. Or just add a `ignoreAnnotated` to its configuration. But I think that, in general, allowing `object A { @JvmStatic val a = 1 }` is a bad idea.

(Thanks @marschwar to point out that there is already a test to check this. That make me think about this problem more deeply)

**Comment by realdadfish** [comment]
You're right, we just needed `@JvmField` for a nullable field there, but not for the non-nullable ones. Sorry for the confusion!
